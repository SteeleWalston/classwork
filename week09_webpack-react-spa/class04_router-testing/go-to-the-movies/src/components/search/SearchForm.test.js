import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import SearchForm from './SearchForm';

describe('SearchForm', () => {

  it('renders as designed', () => {
    const wrapper = shallow(<SearchForm onSearch={() => {}}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders with search term as designed', () => {
    const wrapper = shallow(<SearchForm searchTerm="star wars" onSearch={() => {}}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('calls onSearch with criteria entered', () => {
    let filter;
    const handleFilter = f => {
      filter = f;
    }

    const wrapper = mount(<SearchForm onSearch={handleFilter}/>);

    const search = 'Star Wars';

    // input and click
    wrapper.find('input').simulate('change', {
      target: { value: search }
    })
    wrapper.find('button').simulate('submit');

    // test we got the filter
    expect(filter).toBe(search);

  });

});