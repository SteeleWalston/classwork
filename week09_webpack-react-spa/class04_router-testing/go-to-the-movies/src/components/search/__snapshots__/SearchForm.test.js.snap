// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders as designed 1`] = `
<form
  onSubmit={[Function]}
>
  <fieldset>
    <label>
      Big Search for Movies: 
      <input
        name="search"
        onChange={[Function]}
        placeholder="enter movie search"
        value=""
      />
    </label>
    <label>
       
      <button>
        Search
      </button>
    </label>
  </fieldset>
</form>
`;

exports[`SearchForm renders with search term as designed 1`] = `
<form
  onSubmit={[Function]}
>
  <fieldset>
    <label>
      Big Search for Movies: 
      <input
        name="search"
        onChange={[Function]}
        placeholder="enter movie search"
        value="star wars"
      />
    </label>
    <label>
       
      <button>
        Search
      </button>
    </label>
  </fieldset>
</form>
`;
